src/utils/testRepoManager.ts

```diff
@@ -0,0 +1,120 @@
+import { TEST_REPO_CONFIG } from "../config/config";
+import { exec } from "child_process";
+import util from "util";
+import fs from "fs/promises";
+import path from "path";
+import os from "os";
+import logger from "./logger";
+
+const execAsync = util.promisify(exec);
+
+export class TestRepoManager {
+  private static instance: TestRepoManager;
+  private repoDir: string | null = null;
+  private lastUpdate: number = 0;
+  private readonly updateInterval = 5 * 60 * 1000; // 5 minutes
+
+  private constructor() {
+    // Initialize repoDir to a fixed location
+    this.repoDir = path.join(os.tmpdir(), "hxckr-test-repo");
+  }
+
+  public static getInstance(): TestRepoManager {
+    if (!TestRepoManager.instance) {
+      TestRepoManager.instance = new TestRepoManager();
+    }
+    return TestRepoManager.instance;
+  }
+
+  private async initializeRepo(): Promise<void> {
+    try {
+      // Check if directory exists
+      const exists = await fs
+        .access(this.repoDir!)
+        .then(() => true)
+        .catch(() => false);
+
+      if (!exists) {
+        // First time: Clone the repository
+        await fs.mkdir(this.repoDir!, { recursive: true });
+        await execAsync(
+          `git clone -b ${TEST_REPO_CONFIG.branch} ${TEST_REPO_CONFIG.repoUrl} ${this.repoDir}`,
+        );
+        logger.info("Test repository cloned successfully");
+      }
+    } catch (error) {
+      logger.error("Error initializing test repository", { error });
+      throw error;
+    }
+  }
+
+  private async updateRepo(): Promise<void> {
+    try {
+      await execAsync(`cd ${this.repoDir} && git pull`);
+      logger.info("Test repository updated successfully");
+    } catch (error) {
+      logger.error("Error updating test repository", { error });
+      throw error;
+    }
+  }
+
+  private async ensureRepoUpdated(): Promise<string> {
+    const now = Date.now();
+
+    try {
+      // Initialize repo if it doesn't exist
+      await this.initializeRepo();
+
+      // Update repo if needed
+      if (now - this.lastUpdate > this.updateInterval) {
+        await this.updateRepo();
+        this.lastUpdate = now;
+      }
+
+      return this.repoDir!;
+    } catch (error) {
+      logger.error("Error ensuring repo is updated", { error });
+      throw error;
+    }
+  }
+
+  public async getTestContent(
+    challengeId: string,
+    language: string,
+    stage: number,
+  ): Promise<string> {
+    const repoDir = await this.ensureRepoUpdated();
+    const testPath = path.join(
+      repoDir,
+      TEST_REPO_CONFIG.testsPath,
+      challengeId,
+      language,
+      `stage${stage}.test.ts`,
```

Comment:
This assumes that the test is always in typescript. We should make it dynamic so that it maps the language to the test file extension, similar to how you do it [here](https://github.com/hxckr-org/hxckr-test-runners/pull/3/files#diff-d5f609fba37ce51df609161bf9ca46f3df3f68f9dd631cda374d36e962fc7250R9)

Reply:
Good catch, initially I added it in the referenced file as debug, we are not suppoused to have it there, there already a `getTestExtension` in runner  which is where we need it. 

However, for test, we are not asuming, the equivalent langauage has been accounted for and pass down to there we are checking for its extention.

Reply:
I'm not sure I understand your comment. This code always assumes the test file is in typescript which isn't correct

Reply:
Gothcha

Reply:
fixed